### PR TITLE
feat(generateEmail): add generateEmail

### LINKED ISSUE #21: Converting React components to HTML
### Description:
1. After creating react components, we need to convert the react code to plain HTML code which users can send through email.

2. However, we also need to add the option to configure the base HTML and base CSS styles. Both of them should be added to the separate files and then should be imported. Users can also customize the base code as per their needs.

### Outcome:
1. We need to create the “generateEmail” utility function which will do this conversion.
2. Add stories on how to use the generateEmail utility.

### PR DESCRIPTION
**Pull request by niloysikdar**
### Requirements:

- After creating react components, we need to convert the react code to plain HTML code which users can send through email.

- However, we also need to add the option to configure the base HTML and base CSS styles. Both of them should be added to the separate files and then should be imported. Users can also customize the base code as per their needs.

### The Pull Request covers:

- add generateEmail method to extract plain optimized HTML from the react-based layout
- optional template override options with a default optimized template for baseTemplate
- Passing tests for generateEmail

### Reasons for not using any separate HTML file for default template

- Initially, I also thought the same, but later discovered some weird path issues regarding the function call. That's why I just added it to a simple string.

### Conversion with both default template (passing no options) and with custom base template:
Capture13

### Code for using custom template with another HTML and CSS files:
```
const HTML = generateEmail(SayHello({ name: "Niloy" }), {
  baseTemplate: "./index.html",
});

console.log(HTML);
```

resolves #21

Signed-off-by: Niloy Sikdar <niloysikdar30@gmail.com>

### CODE REVIEW

File: src/server/generateEmail/generateEmail.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  options?.baseTemplate
+    ? (baseHTML = fs.readFileSync(options.baseTemplate, 'utf8')).toString()
+    : (baseHTML = deafultHTML);
+
+  let baseCSS = '';
+  options?.baseStyles && (baseCSS = fs.readFileSync(options.baseStyles, 'utf8')).toString();
+
+  const JSXtoHTML = ReactDOMServer.renderToStaticMarkup(jsxElement);
+
+  const styleEndPos = baseHTML.indexOf('</style>');
```
Comment (agentmilindu) [inline src/server/generateEmail/generateEmail.ts:24]: Let's not pick the `style` element, uses of the library might pass templates which have their own `style` elements. Introduce a variable like 'EMAIL_BASE_STYLES` and `EMAIL_BODY_CONTENT` ( for HTML part ) and replace there with the styles and `JSXtoHTML`

Comment (niloysikdar) [reply src/server/generateEmail/generateEmail.ts]: done

File: src/utils/generateEmail/defaultHTML.ts [hunk 2]
```diff
@@ -0,0 +1 @@
...om:vml" xmlns:o="urn:schemas-microsoft-com:office:office" > <head> <title> </title> <meta http-equiv="X-UA-Compatible" content="IE=edge" /> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /> <meta name="viewport" content="width=device-width, initial-scale=1" /> <style type="text/css"> #outlook a { padding: 0; } body { margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; } table, td { border-collapse: collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt; } img { border: 0; height: auto; line-height: 100%; outline: none; text-decoration: none; -ms-interpolation-mode: bicubic; } p { display: block; margin: 13px 0; } EMAIL_BASE_STYLES </style> <style type="text/css"> @media only screen and (min-width: 480px) { .column-per-100 { width: 100% !important; max-width: 100%; } } </style> <style media="screen and (min-width:480px)"> .moz-text-html .column-per-100 { width: 100% !important; max-width: 100%; } </style> <style type="text/css"> @media only screen and (max-width: 480px) { table.full-width-mobile { width: 100% !important; } td.full-width-mobile { width: auto !important; } } </style> </head> <body> EMAIL_BODY_CONTENT </body> </html>`;
```
Comment (agentmilindu) [inline src/utils/generateEmail/defaultHTML.ts:1]: How about you make this also a `.html` file? Easy for other contributors to read and add contributions to.

Comment (niloysikdar) [reply src/utils/generateEmail/defaultHTML.ts:1]: Initially, I also thought the same, but later discovered some weird path issues regarding the function call. That's why I just added it to a simple string.

**Pull request merged**